lib/config/definitions.ts

```diff
@@ -138,6 +139,16 @@ const options: RenovateOptions[] = [
     default: [],
     cli: false,
   },
+  {
+    name: 'executionMode',
+    description:
+      'Controls whether the post upgrade tasks runs for every dependency or once per upgrade branch',
+    type: 'string',
+    parent: 'postUpgradeTasks',
+    allowedValues: ['dependency', 'branch'],
```

Comment:
```suggestion
    allowedValues: ['update', 'branch'],
```

It's more correctly per-update instead of per-dependency. Potentially there are multiple updates per package within the same branch.